Install issue: Recipe for target 'all' failed

Hi, 

I have been trying to install kmcuda on my system which has Nvidia's GP102 Pascal TitanX and cuda 9.2. But, for some weird reason, I have been facing the following issue when I run the command as: 

`cmake -DCMAKE_BUILD_TYPE=Release -DCUDA_ARCH=61 - DCUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda-9.2 . && make`

And, this is what I get: 

```
[ 33%] Building NVCC (Device) object CMakeFiles/KMCUDA.dir/KMCUDA_generated_kmeans.cu.o
/usr/local/cuda/include/cuda_fp16.hpp(161): error: invalid redeclaration of member function "__half::operator=(const __half_raw &)"
(160): here

/usr/local/cuda/include/cuda_fp16.hpp(162): error: invalid redeclaration of member function "__half::operator=(const __half_raw &)"
(160): here

/usr/local/cuda/include/cuda_fp16.hpp(164): error: invalid redeclaration of member function "__half::operator __half_raw() const"
(163): here

3 errors detected in the compilation of "/tmp/tmpxft_00008f6c_00000000-6_kmeans.cpp1.ii".
CMake Error at KMCUDA_generated_kmeans.cu.o.Release.cmake:279 (message):
  Error generating file
  /mlodata1/sidak/projects/kmcuda_cuda=61/kmcuda/src/CMakeFiles/KMCUDA.dir//./KMCUDA_generated_kmeans.cu.o


CMakeFiles/KMCUDA.dir/build.make:63: recipe for target 'CMakeFiles/KMCUDA.dir/KMCUDA_generated_kmeans.cu.o' failed
make[2]: *** [CMakeFiles/KMCUDA.dir/KMCUDA_generated_kmeans.cu.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/KMCUDA.dir/all' failed
make[1]: *** [CMakeFiles/KMCUDA.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

It would be great if someone can help me on this! Thanks a lot! 😄 

PS: I really like kmcuda's support for fast clustering as well as the angular distance. Now, I have moved to a different machine, and have been stuck in this.

--- a/src/fp_abstraction.h
+++ b/src/fp_abstraction.h
@@ -8,7 +8,11 @@
 #include <cfloat>
 
 #if CUDA_ARCH >= 60
+#undef volatile
+#define volatile volatile
 #include <cuda_fp16.h>
+#undef volatile
+#define volatile
 #endif
 
 #define FPATTR __device__ __forceinline__

